pkg/asset/cluster/tfvars.go

```diff
@@ -425,17 +425,20 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, c := range controlPlanes {
 			controlPlaneConfigs[i] = c.Spec.ProviderSpec.Value.Object.(*vsphereprovider.VSphereMachineProviderSpec)
 		}
-		data, err = vspheretfvars.TFVars(
+		data, cachedImage, err := vspheretfvars.TFVars(
 			vspheretfvars.TFVarsSources{
 				ControlPlaneConfigs: controlPlaneConfigs,
 				Username:            installConfig.Config.VSphere.Username,
 				Password:            installConfig.Config.VSphere.Password,
 				Cluster:             installConfig.Config.VSphere.Cluster,
+				ImageURI:            string(*rhcosImage),
 			},
 		)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get %s Terraform variables", platform)
 		}
+		installConfig.Config.VSphere.ClusterOSImage = cachedImage
```

Comment:
What is the effect of doing this? Is this value saved or used later?

Reply:
The goal here is to stuff the cached location back into the installConfig so we can pass it to the preterraform func. https://github.com/openshift/installer/pull/2922/files#diff-661643e1b9745123649391076bfa5a01R22. I don't believe it's ever written out to disk, though.

Reply:
I don't think you can assume writeback to your dependency assets, the asset store could easily be providing you the ready only copies.